File: src/main/java/com/nurflugel/dependencyvisualizer/enums/Shape.java
package com.nurflugel.dependencyvisualizer.enums;

public enum Shape {
  ellipse,
  rectangle,
  hexagon,
  triangle;

  /** Get the next color in the enum. */
  public static Shape next(Shape shape) {
    Shape[] shapes  = values();
    int     ordinal = shape.ordinal();
    Shape[] values  = values();

    return values[(ordinal + 1) % values.length];
  }

  /** Get the next shape in the enum. */
  public static Shape previous(Shape shape) {
    int     ordinal       = shape.ordinal();
    Shape[] values        = values();
    int     previousIndex = ordinal - 1;

    if (previousIndex < 0) { previousIndex = values.length - 1; }

    return values[previousIndex];
  }

  public static Shape first() { return values()[0]; }

  public static Shape last() { return values()[values().length - 1]; }
}

File: src/main/kotlin/com/nurflugel/dependencyvisualizer/enums/Shape.kt
package com.nurflugel.dependencyvisualizer.enums

enum class Shape {
    box,
    ellipse,
    rectangle,
    hexagon,
    triangle;

    companion object {

        /** Get the next color in the enum.  */
        fun next(shape: Shape): Shape {
            return entries[(shape.ordinal + 1) % entries.size]
        }

        /** Get the next shape in the enum.  */
        fun previous(shape: Shape): Shape {
            val previousIndex = shape.ordinal - 1

            return when {
                previousIndex < 0 -> entries.last()
                else              -> entries[previousIndex]
            }
        }

        //        fun first(): Shape {
        //            return entries[0]
        //        }
        //
        //        fun last(): Shape {
        //            return entries[entries.size - 1]
        //        }
    }
}
